app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthController.java

```diff
@@ -0,0 +1,98 @@
+package gov.hhs.cdc.trustedintermediary.auth;
+
+import gov.hhs.cdc.trustedintermediary.domainconnector.DomainRequest;
+import gov.hhs.cdc.trustedintermediary.domainconnector.DomainResponse;
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import gov.hhs.cdc.trustedintermediary.wrappers.formatter.Formatter;
+import gov.hhs.cdc.trustedintermediary.wrappers.formatter.FormatterProcessingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+/** Parses the request and generates the response that comes in for the auth endpoints. */
+public class AuthController {
+
+    private static final AuthController AUTH_CONTROLLER = new AuthController();
+    static final String CONTENT_TYPE_LITERAL = "Content-Type";
+    static final String APPLICATION_JSON_LITERAL = "application/json";
+
+    @Inject Logger logger;
+    @Inject Formatter formatter;
+
+    private AuthController() {}
+
+    public static AuthController getInstance() {
+        return AUTH_CONTROLLER;
+    }
+
+    public AuthRequest parseAuthRequest(DomainRequest request) throws IllegalArgumentException {
+        logger.logInfo("Parsing login request via JWT");
+
+        Map<String, Optional<String>> authFields = extractFormUrlEncode(request.getBody());
+
+        return new AuthRequest(
+                authFields.getOrDefault("scope", Optional.empty()).orElse(null),
+                authFields.getOrDefault("client_assertion", Optional.empty()).orElse(null));
+    }
+
+    public DomainResponse constructResponse(int httpStatus) {
+        return constructResponse(httpStatus, null);
+    }
+
+    public DomainResponse constructResponse(int httpStatus, String payload) {
+
+        DomainResponse response = new DomainResponse(httpStatus);
+
+        if (payload != null) {
+            response.setHeaders(Map.of(CONTENT_TYPE_LITERAL, APPLICATION_JSON_LITERAL));
+            response.setBody(payload);
+        }
+
+        return response;
+    }
+
+    public String constructPayload(AuthRequest authRequest, String token)
+            throws FormatterProcessingException {
+
+        String payloadJson;
+        String bearerToken = "";
+
+        if (token != null && !token.isBlank()) {
```

Comment:
We could.  I'm not the biggest fan of ternary operators, but whatever.  I think we could also just do away with the entire if statement and put `token` directly into the `payload` map.

Reply:
There are two possible conditions to check for. 
1. Is the string null
2. If is is not null, is the the string blanked with whitespaces

If we want to use a ternary, then it would be something like:

`String bearerToken = (token != null) ? (!token.isBlank() ? token: "") : ""` 
or
`String bearerToken = (token == null) ? "" : (token.isBlank() ? "" : token)`

One thing to consider is, should we be checking for blank tokens? thoughts?

Reply:
We decided on the nice ternary. Good call Basilio!